ibis/client.py

```diff
@@ -923,6 +923,136 @@ def _get_schema_using_query(self, query):
 
         return ir.Schema(names, ibis_types)
 
+    def create_udf(self, udf_info, udf_name=None, db=None):
+        """
+        Creates a function within Impala
+
+        Parameters
+        ----------
+        udf_info : UDFInfo object
+        udf_name : string (optional)
+        db : string (optional)
+        """
+        name = udf_info.get_name()
+        if udf_name:
+            name = udf_name
+        statement = ddl.CreateFunction(udf_info.hdfs_file,
+                                       udf_info.so_symbol,
+                                       udf_info.inputs,
+                                       udf_info.output,
+                                       name, db)
+        self._execute(statement)
+
+    def create_uda(self, uda_info, uda_name=None, db=None):
+        """
+        Creates a user-defined aggregate function within Impala
+
+        Parameters
+        ----------
+        uda_info : UDAInfo object
+        uda_name : string (optional)
+        db : string (optional)
+        """
+        name = uda_info.get_name()
+        if uda_name:
+            name = uda_name
+        statement = ddl.CreateAggregateFunction(uda_info.hdfs_file,
+                                                uda_info.input_type,
+                                                uda_info.output_type,
+                                                uda_info.init_fn,
+                                                uda_info.update_fn,
+                                                uda_info.merge_fn,
+                                                uda_info.finalize_fn,
+                                                name,
+                                                db=db)
+        self._execute(statement)
+
+    def drop_udf(self, name, input_types, db=None, must_exist=False,
```

Comment:
1) Can we make `input_types` optional, and, if not provided, use the information from `SHOW FUNCTIONS` to drop the function

2) the other `drop_*` functions use `force` instead of `must_exist`

3) what do you think about dropping all overloaded instances of a function name with `input_types=None` and `force=True`? perhaps use this case to complete #290